### PR TITLE
Finalizing 1.3 release

### LINKED ISSUE #13501: Add missing `AdderGate` class or remove references
### What should we add?

The 1.3.0 release notes referred to an `AdderGate` class, described as: "`AdderGate` implements standard addition including a carry-out" and this is also referred to in several docstrings for specific adder classes. But I have found no class definition for `AdderGate`. Either we should add the missing class or remove the references (I removed the release note for 1.3.0 in #13447 already)

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Fix stray links in the releasenotes and the link to `HalfAdderGate` (fixes #13501).

### CODE REVIEW

File: releasenotes/notes/1.3/dag-oxide-60b3d7219cb21703.yaml [hunk 1]
```diff
@@ -28,8 +28,8 @@ upgrade_transpiler:
 
     however this type of mutation was always unsound as it could break the DAG's internal caching
     and cause corruption of the data structure. Instead you should use the API provided by
-    :class:`.DAGCircuit` for mutation such as :meth:`.DAGCircuit.substitute_node`,
-    :meth:`.DAGCircuit.substitute_node_with_dag`, or :meth:`.DAGCircuit.contract_node`. For example
```
Comment (alexanderivrii) [inline releasenotes/notes/1.3/dag-oxide-60b3d7219cb21703.yaml:32]: What is the reason for removing this from the release notes? Update: I did not copy the correct lines, only the `contract_nodes` method got removed.

**Review by alexanderivrii — DISMISSED** [review]
All of these polishing changes make sense to me. Special thanks for fixing the documentation for `HalfAdderGate`s

File: releasenotes/notes/1.3/dag-oxide-60b3d7219cb21703.yaml (hunk 1 above)
Comment (Cryoris) [reply releasenotes/notes/1.3/dag-oxide-60b3d7219cb21703.yaml:32]: `DAGCircuit.contract_node` does not exist 🙂 That was probably a confusion coming from petgraph's `StableGraph::contract_node`, but this is not exposed on the `DAGCircuit`.

**Comment by eliarbel** [comment]
Good catches @Cryoris and thanks for fixing these!

There is one more stray link in https://github.com/Qiskit/qiskit/blob/cda7ef635678a17e8041a844a6712da06435da2a/releasenotes/notes/1.3/oxidize-transpiler-passes-16f2f1c57612e74a.yaml#L22. Can you please fix as well?

File: releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml [hunk 2]
```diff
@@ -34,7 +34,6 @@ deprecations_qpy:
 deprecations_transpiler:
   - |
     As part of the Qiskit Pulse package deprecation, pulse-related aspects in the :class:`qiskit.transpiler.Target` class are being deprecated. These include:
-      * :attr:`~qiskit.transpiler.Target.calibration`
```
Comment (eliarbel) [inline releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:37]: Why removing this?

Comment (Cryoris) [reply releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:37]: There's no attribute (or method) called `calibration` 🙂

Comment (mtreinish) [reply releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:37]: It's `InstructionProperties.calibration`: https://docs.quantum.ibm.com/api/qiskit/qiskit.transpiler.InstructionProperties#calibration probably.

Comment (eliarbel) [reply releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml:37]: Oh, yes, it's in `InstructionProperties` and covered already 😄

File: releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml [hunk 3]
```diff
@@ -47,15 +46,15 @@ deprecations_transpiler:
...
   - |
-    The ``inst_map`` argument in :func:`~qiskit.transpiler.generate_preset_pass_manager`,
-    :meth:`~transpiler.target.Target.from_configuration`, :class:`~qiskit.transpiler.PassManagerConfig` initializer
+    The ``inst_map`` argument in :func:`.generate_preset_pass_manager`,
+    :meth:`~.Target.from_configuration`, :class:`~qiskit.transpiler.PassManagerConfig` initializer
     and :func:`~qiskit.transpiler.preset_passmanagers.common.generate_scheduling` is being deprecated.
   - |
     The ``calibration`` argument in :func:`~qiskit.transpiler.target.InstructionProperties` initializer methods is being
     deprecated.
 deprecations_visualization:
```
Comment (eliarbel) [inline releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml]: Let's remove this section altogether. `pulse_drawer` is not part of the public API. Furthermore we're already mentioning in the general deprecation note that pulse visualization is also being deprecated and attempts to visualize pulse will raise deprecation warnings of pulse classes/builder functions anyway.

Comment (Cryoris) [reply releasenotes/notes/1.3/deprecate-pulse-package-07a621be1db7fa30.yaml]: done in 925f0e7

**Review by eliarbel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**